src/com/magento/idea/magento2plugin/lang/php/MagentoProxyDeclarationFilter.java

```diff
@@ -47,7 +50,23 @@ public <E extends PhpNamedElement> Collection<E> filter(
                         return false;
                     }
 
-                    return !virtualFile.getPath().contains("/generated/");
+                    return !virtualFile.getPath().contains("/generated/") && isValidFqn(candidate);
                 });
     }
+
+    private <E extends PhpNamedElement> boolean isValidFqn(final E candidate) {
+        PhpClass targetClass = null;
+
+        if (candidate instanceof Method) {
+            targetClass = (PhpClass) candidate.getParent();
+        } else if (candidate instanceof PhpClass) {
+            targetClass = (PhpClass) candidate;
+        }
+
+        if (targetClass == null) {
+            return true;
+        }
+
+        return PhpLangUtil.isFqn(targetClass.getFQN()) && !targetClass.getFQN().endsWith("\\");
```

Comment:
Why we don’t use here `com.magento.idea.magento2plugin.actions.generation.generator.util.PhpClassGeneratorUtil#isValidFqn`?

Reply:
Hello, @VitaliyBoyko, thank you for review, 
I didn't use **com.magento.idea.magento2plugin.actions.generation.generator.util.PhpClassGeneratorUtil#isValidFqn** , because this method returns true when checking **Fqn** that ends with '**\\**'

Thank you.